packages/headless/src/app/undoable.ts

```diff
@@ -54,24 +54,20 @@ const redo = <State>(state: StateWithHistory<State>) => {
   };
 };
 
-const updateHistory = <State, Action extends AnyAction>(
-  state: StateWithHistory<State>,
-  emptyState: StateWithHistory<State>,
-  reducer: Reducer<State>,
-  action: Action
-) => {
+const updateHistory = <State>(options: {
+  state: StateWithHistory<State>;
+  reducer: Reducer<State>;
+  action: AnyAction;
+}) => {
+  const {action, state, reducer} = options;
   const {past, present} = state;
   const newPresent = reducer(present, action);
-
-  if (newPresent === emptyState.present || present === newPresent) {
-    return state;
+  if (!present) {
+    return makeHistory(newPresent);
   }
 
-  // Small special twist on the documented/standard redux undo recipe
-  // We want to make the "actual first valid initial state" the first one that ends up being different from the "empty state" passed into undoable
-  // This allows for slices to register themselves dynamically (Concrete example: facet-slice).
-  if (present === emptyState.present) {
```

Comment:
Special twist made history only work correctly when facets were present. When facets weren't present, the second search was entering this condition and not adding the past correctly. Only the after the 3rd search it worked as expected.
When removing this condition, adding facets was updating history, hence why the twist was added in the first place.

Reply:
Why facets? because they were actively modifying the historyState on registration, meanwhile for most component, registration is equal to the empty state.